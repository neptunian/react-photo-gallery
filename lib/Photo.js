'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.photoPropType = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var imgWithClick = { cursor: 'pointer' };

var Photo = function (_PureComponent) {
  _inherits(Photo, _PureComponent);

  function Photo() {
    _classCallCheck(this, Photo);

    var _this = _possibleConstructorReturn(this, (Photo.__proto__ || Object.getPrototypeOf(Photo)).call(this));

    _this.handleClick = _this.handleClick.bind(_this);
    return _this;
  }

  _createClass(Photo, [{
    key: 'handleClick',
    value: function handleClick(event) {
      var _props = this.props,
          onClick = _props.onClick,
          index = _props.index,
          photo = _props.photo;

      onClick(event, { photo: photo, index: index });
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props,
          photo = _props2.photo,
          onClick = _props2.onClick,
          margin = _props2.margin;

      var imgStyle = { display: 'block', float: 'left', margin: margin };
      return _react2.default.createElement('img', _extends({
        style: onClick ? _extends({}, imgStyle, imgWithClick) : imgStyle
      }, photo, {
        src: photo.thumbnail ? photo.thumbnail : photo.src,
        onClick: onClick ? this.handleClick : null
      }));
    }
  }]);

  return Photo;
}(_react.PureComponent);

var photoPropType = exports.photoPropType = _propTypes2.default.shape({
  src: _propTypes2.default.string.isRequired,
  width: _propTypes2.default.number.isRequired,
  height: _propTypes2.default.number.isRequired,
  alt: _propTypes2.default.string,
  title: _propTypes2.default.string,
  srcSet: _propTypes2.default.array,
  sizes: _propTypes2.default.array,
  thumbnail: _propTypes2.default.string
});

Photo.propTypes = {
  index: _propTypes2.default.number,
  onClick: _propTypes2.default.func,
  photo: photoPropType
};

exports.default = Photo;