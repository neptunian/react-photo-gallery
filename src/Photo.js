import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

const imgWithClick = { cursor: 'pointer' };

class Photo extends PureComponent {
  constructor() {
    super();
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(event) {
    const { onClick, index, photo } = this.props;
    onClick(event, { photo, index });
  }

  render() {
    const { photo, onClick, margin } = this.props;
    const imgStyle = { display: 'block', float: 'left', margin: margin };
    return (
      <img
        style={onClick ? { ...imgStyle, ...imgWithClick } : imgStyle}
        {...photo}
        src={photo.thumbnail ? photo.thumbnail : photo.src}
        onClick={onClick ? this.handleClick : null}
      />
    );
  }
}

export const photoPropType = PropTypes.shape({
  src: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  alt: PropTypes.string,
  title: PropTypes.string,
  srcSet: PropTypes.array,
  sizes: PropTypes.array,
  thumbnail: PropTypes.string,
});

Photo.propTypes = {
  index: PropTypes.number,
  onClick: PropTypes.func,
  photo: photoPropType,
};

export default Photo;
