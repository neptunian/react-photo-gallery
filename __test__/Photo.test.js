import React from 'react';
import { render, shallow } from 'enzyme';

import Photo from '../src/Photo';

describe('Photo component', () => {
  it('renders correctly with onClick', () => {
    const mockCallBack = jest.fn();

    const photo = {
      src: 'http://example.com/photo.jpg',
      width: 50,
      height: 50,
    };

    const component = shallow(<Photo photo={photo} onClick={mockCallBack} />);
    component.find('img').simulate('click');

    expect(component).toMatchSnapshot();
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('renders correctly without onClick', () => {
    const photo = {
      src: 'http://example.com/photo.jpg',
      width: 50,
      height: 50,
    };

    const component = render(<Photo photo={photo} />);
    expect(component).toMatchSnapshot();
  });

  it('should display the thumbnail', () => {
    const photo = {
      src: 'http://example.com/photo.jpg',
      thumbnail: 'http://example.com/thumbnail.jpg',
      width: 50,
      height: 50,
    };

    const component = shallow(<Photo photo={photo} />);
    const img = component.find('img').filterWhere(item => {
      return item.prop('src') === photo.thumbnail;
    });
    expect(img).toHaveLength(1);
  });
});
