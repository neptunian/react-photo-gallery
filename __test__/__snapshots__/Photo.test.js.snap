// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo component renders correctly with onClick 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Photo
    onClick={[Function]}
    photo={
      Object {
        "height": 50,
        "src": "http://example.com/photo.jpg",
        "width": 50,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "height": 50,
      "onClick": [Function],
      "src": "http://example.com/photo.jpg",
      "style": Object {
        "cursor": "pointer",
        "display": "block",
        "float": "left",
        "margin": undefined,
      },
      "width": 50,
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "height": 50,
        "onClick": [Function],
        "src": "http://example.com/photo.jpg",
        "style": Object {
          "cursor": "pointer",
          "display": "block",
          "float": "left",
          "margin": undefined,
        },
        "width": 50,
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Photo component renders correctly without onClick 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "height": "50",
      "src": "http://example.com/photo.jpg",
      "style": "display:block;float:left",
      "width": "50",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "height": undefined,
      "src": undefined,
      "style": undefined,
      "width": undefined,
    },
    "x-attribsPrefix": Object {
      "height": undefined,
      "src": undefined,
      "style": undefined,
      "width": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
